!in does not compile when used with maps

If you try using the newly implemented `!in` operator on a map, you'll get a C compile time error.

This is what I get in REPL. In real code the error seems to be different.
```
For usage information, quit V REPL using `exit` and use `v help`
V 0.1.26 83289d7.6ada43d
Use Ctrl-C or `exit` to exit
>>> m := { 'a' : 1, 'b' : 2 }
>>> 'a' in m
true
>>> 'a' !in m
==================
/home/andole/.cache/v/.noprefix.vrepl_temp.tmp.c: In function ‘main’:
/home/andole/.cache/v/.noprefix.vrepl_temp.tmp.c:4627:33: error: incompatible type for argument 1 of ‘bool_str’
 4627 |  println(bool_str(/*node error*/tos3("a"), m)));
      |                                 ^~~~~~~~~
      |                                 |
      |                                 string {aka struct string}
/home/andole/.cache/v/.noprefix.vrepl_temp.tmp.c:2309:22: note: expected ‘bool’ {aka ‘int’} but argument is of type ‘string’ {aka ‘struct string’}
 2309 | string bool_str(bool b) {
      |                 ~~~~~^
/home/andole/.cache/v/.noprefix.vrepl_temp.tmp.c:4627:10: error: too many arguments to function ‘bool_str’
 4627 |  println(bool_str(/*node error*/tos3("a"), m)));
      |          ^~~~~~~~
...
==================
```

--- a/vlib/v/gen/cgen.v
+++ b/vlib/v/gen/cgen.v
@@ -1336,7 +1336,7 @@ fn (g mut Gen) infix_expr(node ast.InfixExpr) {
 	// g.infix_op = node.op
 	left_sym := g.table.get_type_symbol(node.left_type)
 	right_sym := g.table.get_type_symbol(node.right_type)
-	if node.left_type == table.string_type_idx && node.op != .key_in {
+	if node.left_type == table.string_type_idx && node.op != .key_in && node.op != .not_in {
 		fn_name := match node.op {
 			.plus {
 				'string_add('

